llvm/lib/Target/AIE/AIE2LegalizerInfo.cpp

```diff
@@ -461,6 +461,17 @@ AIE2LegalizerInfo::AIE2LegalizerInfo(const AIE2Subtarget &ST) : AIEHelper(ST) {
       .clampMaxNumElements(0, S32, 16)
       .custom();
 
+  getActionDefinitionsBuilder(G_BUILD_VECTOR_TRUNC)
```

Comment:
Do you know how those `G_BUILD_VECTOR_TRUNC` get generated and why we need to support them only now?

Reply:
We need `G_BUILD_VECTOR_TRUNC` because input element for VInsert.8 is `s32`, if we use existing `G_BUILD_VECTOR` we need to build truncations that truncates `s32` into `s8`.

Reply:
Makes sense, better avoid extra tunc/ext :)